internal/resolution/resolver.go

```diff
@@ -58,10 +85,61 @@ func (r Resolver) setNpmPreferred(npmPreferred bool) {
 	r.batchFactory.SetNpmPreferred(npmPreferred)
 }
 
+func (r Resolver) GetExitCode(resolution IResolution, options IOptions) (int, error) {
+	dOptions, ok := options.(DebrickedOptions)
+	if !ok {
+		return 1, ErrBadOpts
+	}
+	errorCount := resolution.GetJobErrorCount()
+	jobCount := len(resolution.Jobs())
+
+	return r.getExitCodeBasedOnStrictness(dOptions.Resolutionstrictness, errorCount, jobCount)
+}
+
+type exitCodeLogic func(errorCount, jobCount int) (int, error)
+
+func (r Resolver) getExitCodeBasedOnStrictness(strictness StrictnessLevel, errorCount, jobCount int) (int, error) {
+	exitCodeLogics := map[StrictnessLevel]exitCodeLogic{
+		NoFail: func(errorCount, jobCount int) (int, error) {
+			return 0, nil
+		},
+		FailIfAllFail: func(errorCount, jobCount int) (int, error) {
+			if errorCount == jobCount {
+				return 1, nil
+			}
+
+			return 0, nil
+		},
+		FailIfAnyFail: func(errorCount, jobCount int) (int, error) {
+			if errorCount > 0 {
+				return 1, nil
+			}
+
+			return 0, nil
+		},
+		FailOrWarn: func(errorCount, jobCount int) (int, error) {
+			if errorCount == 0 {
+				return 0, nil
+			} else if errorCount == jobCount {
+				return 1, nil
+			}
+
+			return 3, nil
+		},
+	}
+
+	logic, ok := exitCodeLogics[strictness]
```

Comment:
This looks like a case-statement with extra steps, could we change it to that to reduce complexity or am I missing something here?